### PR TITLE
docs(knowledge-base): add nextauth flow to docs

### PR DESCRIPTION
**Pull request by AquaShotRyan**
Screenshot 2024-08-19 at 11-56-00 NextAuth js Flow

In the outline on the right side, there's also this:
image

### CODE REVIEW

File: apps/docs/src/content/docs/knowledge-base/nextauth.mdx [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        id: ctx.session.user.id,
+      },
+    });
+    return user;
+  }),
+});
+```
+
+
+## 📖Sources and Further Reading
```
Comment (HasithDeAlwis) [inline apps/docs/src/content/docs/knowledge-base/nextauth.mdx:435]: Nice touch with the sources and further reading 🔥

File: apps/docs/src/content/docs/knowledge-base/nextauth.mdx [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    callbacks: {
+      authorized: ({ token }) => token?.role === "admin",
+    },
+  },
+)
+
+export const config = { matcher: ["/admin"] }
+```
+
+## 🛡️Another Method for Protection and Authentication
```
Comment (HasithDeAlwis) [inline apps/docs/src/content/docs/knowledge-base/nextauth.mdx:332]: Could we just rename it to General Method for Protection and Authentication?

**Review by HasithDeAlwis — DISMISSED** [review]
🔥 🔥

**Review by MFarabi619 — APPROVED** [review]
Nice, very neat. Huge improvement from the last one. Approved. Small nitpicks, completely up to you if you want to implement them in this PR or a different one.

- Your edit for Hasith's commit can be amended to your initial commit and force pushed.
    - This keeps the history a bit cleaner.
    - I make a new commit if it's something I'm changing that already existed in the main branch.
    - Since this is new content, I'd amend it since it's 'what's supposed to go into main to begin with'.
    - This comes with practice, so don't worry too much about it.
- This outline is the Table of Contents, it will always exist in every page and is auto-generated based on the headings and sub-headings in your `.mdx` document, so you don't have to include this in your PR description.

image
    - However if you had made a change that affected its default behaviour, for example changing the text colour or background color of that component, then yes you'd include it.
- There's a whole bunch of stuff you can do with tabs, colors, and other stuff to make it easier for people to interact with your documentation.
    - Your docs will 'flow' much better and read less like a wall of text. You can browse other areas of the docs site and see the corresponding source code for examples.
    - However don't worry about them too much for now, unless you're bored and want to learn. Completely optional. 
    
Migrations aren't easy. Glad we got this off Notion and on the Docs site. 🙏 
Also congrats on your first PR merged (soon) 🥳

**Review by HasithDeAlwis — APPROVED** [review]

**Pull request merged**